### PR TITLE
[amazon] fixes for DE

### PR DESCRIPTION
**Pull request by moritzj29**
Hi, there have been some minor changes on Amazon DE webpage, which broke the invoice scraping. Also I fixed handling archived orders for non EN languages.

### CODE REVIEW

File: finance_dl/amazon.py [hunk 1]
```diff
@@ -402,25 +402,21 @@ def invoice_link_finder(invoice_link):
...
                     return (order_id, href)
 
-                def invoice_link_finder_hidden():
-                        # submenu containing order summary takes some time to load after click
-                        # search for order summary link and compare order_id
-                        # repeat until order_id is different to last order_id
-                        summary_links = self.driver.find_elements(By.LINK_TEXT, 
-                            self.domain.order_summary)
-                        if summary_links:
-                            href = summary_links[0].get_attribute('href')
```
Comment (Zburatorul) [inline finance_dl/amazon.py:408]: What's the logic for inserting this code into directly the else clause? It seems more readable they way it was before.

Comment (moritzj29) [reply finance_dl/amazon.py:408]: you are right, should be fixed now

**Comment by Zburatorul** [comment]
@moritzj29 could you resolve the conflicts please?

**Comment by moritzj29** [comment]
Sorry for the delayed response, unfortunately there have been issues again with the link extraction for hidden invoices (DE), which should be fixed now.

Regarding the conflict:

```
<<<<<<< dev/amazon
                (order_filter,), = self.wait_and_return(
                    lambda: self.find_visible_elements(By.XPATH, '//select[@name="timeFilter"]')
                )
=======
                order_filter, = self.wait_and_locate((By.CSS_SELECTOR, '#time-filter, #orderFilter'))
>>>>>>> master
```
#97 introduced getting the order filter by a `CSS_SELECTOR`. In my case, this does not work reliably in headless mode. I myself adjusted the existing `XPATH` to the updated page. I have no issues with `XPATH`, also in headless mode. Therefore I would argue for going with this original approach. Please let me know if you think otherwise.

**Comment by Zburatorul** [comment]
Tested. Works.

**Pull request merged**